update.sh

```diff
@@ -8,44 +8,52 @@
 
 set -e
 
+# Check whether there is a newer version of this script
+cd /usr/share/rhino/rhino-update
+if git pull; then
+    chmod +x /usr/share/rhino/rhino-update/update.sh
+    exec /usr/bin/rhino-update
+    exit $?
+fi
+
```

Comment:
This looks like a good start to check for a new version of a script. I believe it's checking if git pull can be executed it will mark it executable and then run the script? My question is that will it still run even if the script hasn't been updated, for things such as package upgrades?

Reply:
well yes it will still run. So for example a user runs `rhino-update`. Git pull is ran and there is new changes, say, a new kernel, then the script would rerun itself with the new version. In the new version we already git pulled the new changes so the script runs the rest of itself. I think this if statement may need to be changed though to make sure there are actual new changes though.